docs/en/protocols/HTTP-API-Protocol.md

```diff
@@ -0,0 +1,198 @@
+# HTTP API Protocol
+
+HTTP API Protocol define the API data format, including api request and response data format.
+
+### Do register
+
+For more information about data format can be found in [Register service](https://github.com/apache/skywalking-data-collect-protocol/tree/master/register/Register.proto).
+And register steps followings [SkyWalking Trace Data Protocol v2](Trace-Data-Protocol-v2.md).
+
+- Service Register
+
+> POST http://localhost:12800/v2/service/register
+
+Input:
+
+```json
+{
+  "services": [
+    {
+      "type": "normal",
+      "serviceName": "Service Name"
+    }
+  ]
+}
+```
+
+Output JSON Array:
+
+```json
+[
+    {
+        "key": "Service Name",
+        "value": 2
+    }
+]
+```
+
+- Service instance Register
+
+> POST http://localhost:12800/v2/instance/register
+
+Input:
+
+```json
+{
+  "instances": [
+    {
+      "time": 1582428603392,
+      "instanceUUID": "NAME:Service Instance Name",
+      "properties": [
+        {
+          "key": "language",
+          "value": "Lua"
+        }
+      ],
+      "serviceId": 2
+    }
+  ]
+}
+```
+
+OutPut:
+
+```json
+{
+    "key": "NAME:Service Instance Name",
+    "value": 2
+}
```

Comment:
It should be an array, right？

Reply:
https://github.com/apache/skywalking-nginx-lua/blob/e284313e9186ad84d9c68c0042aa55139469fa4b/lib/skywalking/client.lua#L146-L153

This is the code that `skywalking-nginx-lua` handles the response.